tasty-query/shared/src/main/scala/tastyquery/Trees.scala

```diff
@@ -638,7 +654,27 @@ object Trees {
       extends TypeTree(span) {
 
     override protected def calculateType(using Context): Type =
-      underlying.toType // TODO Actually take the refinements into account
+      val base = underlying.toType
+      refinements.foldLeft(base) { (parent, refinement) =>
+        refinement match
+          case TypeMember(name, rhs, _) =>
+            val refinedBounds = rhs match
+              case TypeAliasDefinitionTree(tpt) =>
+                TypeAlias(tpt.toType)
+              case rhs: TypeBoundsTree =>
+                rhs.toTypeBounds
+              case _ =>
+                throw InvalidProgramStructureException(s"Unexpected rhs for type refinement '$name': $rhs")
+            end refinedBounds
+            TypeRefinement(parent, name, refinedBounds)
+
+          case ValDef(name, tpt, _, _) =>
+            TermRefinement(parent, name, tpt.toType)
+
+          case DefDef(name, paramClauses, resultType, _, _) =>
+            TermRefinement(parent, name, ParamsClause.makeDefDefType(paramClauses, resultType))
```

Comment:
I believe that this does not distinguish `Foo { val x: Int }` from `Foo { def x: Int }`, in the future we may want to add an api here for consumers to tell the difference (e.g. for detecting "caching" opportunities)